platform/kube/inject/inject.go

```diff
@@ -446,6 +446,13 @@ func intoObject(c *Config, in interface{}) (interface{}, error) {
 		return nil, err
 	}
 
+	// skip special kubernetes system namespaces
+	for _, namespace := range ignoredNamespaces {
+		if obj.GetNamespace() == namespace {
+			return out, nil
+		}
+	}
+
```

Comment:
This is just moving from initializer to kube-inject.. is it because initializer calls into inject.go ?

Reply:
Yes. Resources in istio-system need their pending initializer list to be cleared without the sidecar actually being injected.